webdriver_factory issue - build_firefox raises exception when using GeckoDriverManager

Hi, we found an issue with `webdriver_factory` that has been introduced in 1.14.0 by this commit:
https://github.com/ElSnoMan/pyleniumio/commit/737a6f17a447bc8290374c6a2c41275b26d51756#diff-a0691ef19635b2f2655886d510eb28feefd3b797e34a732384df81386f67e303R267
`GeckoDriverManager`'s install method returns the driver path (e.g. `/Users/mariuszrusiniak/.wdm/drivers/geckodriver/macos/v0.30.0/geckodriver`) which is the passed to `webdriver.Firefox` as  positional argument which lands in `firefox_profile` keyword argument. It is later on checked if it is a directory and raises an Exception like below:
```
venv/lib/python3.9/site-packages/pylenium/driver.py:265: in visit
    self.webdriver.get(url)
venv/lib/python3.9/site-packages/pylenium/driver.py:233: in webdriver
    return self.init_webdriver() if self._webdriver is None else self._webdriver
venv/lib/python3.9/site-packages/pylenium/driver.py:203: in init_webdriver
    self._webdriver = webdriver_factory.build_from_config(self.config)
venv/lib/python3.9/site-packages/pylenium/webdriver_factory.py:133: in build_from_config
    return build_firefox(
venv/lib/python3.9/site-packages/pylenium/webdriver_factory.py:278: in build_firefox
    return webdriver.Firefox(GeckoDriverManager(version=version).install(), options=options, **(webdriver_kwargs or {}))
venv/lib/python3.9/site-packages/selenium/webdriver/firefox/webdriver.py:126: in __init__
    self.binary = None
venv/lib/python3.9/site-packages/selenium/webdriver/firefox/firefox_profile.py:82: in __init__
    shutil.copytree(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

src = '/Users/mariuszrusiniak/.wdm/drivers/geckodriver/macos/v0.30.0/geckodriver', dst = '/var/folders/0k/d1yrv3sx1ldb_dv2dpzcwkxw0000gn/T/tmp7jfodd7e/webdriver-py-profilecopy', symlinks = False
ignore = <function ignore_patterns.<locals>._ignore_patterns at 0x12457d3a0>, copy_function = <function copy2 at 0x102e2c1f0>, ignore_dangling_symlinks = False, dirs_exist_ok = False

    def copytree(src, dst, symlinks=False, ignore=None, copy_function=copy2,
                 ignore_dangling_symlinks=False, dirs_exist_ok=False):
        """Recursively copy a directory tree and return the destination directory.
    
        dirs_exist_ok dictates whether to raise an exception in case dst or any
        missing parent directory already exists.
    
        If exception(s) occur, an Error is raised with a list of reasons.
    
        If the optional symlinks flag is true, symbolic links in the
        source tree result in symbolic links in the destination tree; if
        it is false, the contents of the files pointed to by symbolic
        links are copied. If the file pointed by the symlink doesn't
        exist, an exception will be added in the list of errors raised in
        an Error exception at the end of the copy process.
    
        You can set the optional ignore_dangling_symlinks flag to true if you
        want to silence this exception. Notice that this has no effect on
        platforms that don't support os.symlink.
    
        The optional ignore argument is a callable. If given, it
        is called with the `src` parameter, which is the directory
        being visited by copytree(), and `names` which is the list of
        `src` contents, as returned by os.listdir():
    
            callable(src, names) -> ignored_names
    
        Since copytree() is called recursively, the callable will be
        called once for each directory that is copied. It returns a
        list of names relative to the `src` directory that should
        not be copied.
    
        The optional copy_function argument is a callable that will be used
        to copy each file. It will be called with the source path and the
        destination path as arguments. By default, copy2() is used, but any
        function that supports the same signature (like copy()) can be used.
    
        """
        sys.audit("shutil.copytree", src, dst)
>       with os.scandir(src) as itr:
E       NotADirectoryError: [Errno 20] Not a directory: '/Users/mariuszrusiniak/.wdm/drivers/geckodriver/macos/v0.30.0/geckodriver'

../../.pyenv/versions/3.9.9/lib/python3.9/shutil.py:564: NotADirectoryError

```

The fix here would be to revert this line to become:
```
return webdriver.Firefox(executable_path=GeckoDriverManager(version=version).install(), options=options, **(webdriver_kwargs or {}))
```

As this is basically a oneliner let me know if you want me to open a PR for this one. Thanks!

--- a/pylenium/webdriver_factory.py
+++ b/pylenium/webdriver_factory.py
@@ -275,7 +275,7 @@ def build_firefox(
         return webdriver.Firefox(
             executable_path=local_path, options=options, capabilities=caps, **(webdriver_kwargs or {})
         )
-    return webdriver.Firefox(GeckoDriverManager(version=version).install(), options=options, **(webdriver_kwargs or {}))
+    return webdriver.Firefox(executable_path=GeckoDriverManager(version=version).install(), options=options, **(webdriver_kwargs or {}))
 
 
 def build_ie(

--- a/tests/unit/test_bugfixes.py
+++ b/tests/unit/test_bugfixes.py
@@ -1,0 +1,14 @@
+from unittest.mock import patch, MagicMock
+from pylenium import webdriver_factory
+
+
+def test_235_webdriver_factory(tmpdir):
+    file_to_use = tmpdir / "file"
+    file_to_use.write("test")
+    gecko_mock = MagicMock()
+    gecko_mock.install.return_value = file_to_use
+    with patch("pylenium.webdriver_factory.GeckoDriverManager", return_value=gecko_mock):
+        with patch("pylenium.webdriver_factory.webdriver.firefox.webdriver.Service"):
+            with patch("pylenium.webdriver_factory.webdriver.firefox.webdriver.RemoteWebDriver"):
+                webdriver_factory.build_firefox("latest", ["headless"], {}, [], [], "", {})
+                gecko_mock.install.assert_called_once()

